apps/omg_watcher/test/integration/invalid_exit_test.exs

```diff
@@ -0,0 +1,159 @@
+# Copyright 2018 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.Integration.InvalidExitTest do
+  use ExUnitFixtures
+  use ExUnit.Case, async: false
+  use OMG.API.Fixtures
+  use OMG.API.Integration.Fixtures
+  use Plug.Test
+  use Phoenix.ChannelTest
+
+  alias OMG.API
+  alias OMG.Eth
+  alias OMG.JSONRPC.Client
+  alias OMG.Watcher.Eventer.Event
+  alias OMG.Watcher.Integration.TestHelper, as: IntegrationTest
+  alias OMG.Watcher.Web.Channel
+
+  @moduletag :integration
+
+  @timeout 40_000
+  @eth API.Crypto.zero_address()
+
+  @endpoint OMG.Watcher.Web.Endpoint
+
+  @tag fixtures: [:watcher_sandbox, :stable_alice, :child_chain, :token, :stable_alice_deposits]
+  test "transaction which is using already spent utxo from exit and happened before end of m_sv causes to emit invalid_exit event ",
+       %{stable_alice: alice, stable_alice_deposits: {deposit_blknum, _}} do
+    # TODO remove this tx , use directly deposit_blknum to get_exit_data
+    tx = API.TestHelper.create_encoded([{deposit_blknum, 0, 0, alice}], @eth, [{alice, 10}])
+    {:ok, %{blknum: deposit_blknum}} = Client.call(:submit, %{transaction: tx})
+
+    bad_tx = API.TestHelper.create_recovered([{deposit_blknum, 0, 0, alice}], @eth, [{alice, 10}])
+
+    %{hash: bad_block_hash, number: bad_block_number, transactions: _} =
+      bad_block = API.Block.hashed_txs_at([bad_tx], 27_000)
+
+    {:module, BadChildChainBLock, _, _} = OMG.Watcher.Integration.BadChildChainBLock.create_module(bad_block)
+
+    JSONRPC2.Servers.HTTP.http(BadChildChainBLock, port: BadChildChainBLock.port())
+
+    {:ok, _, _socket} = subscribe_and_join(socket(), Channel.Byzantine, "byzantine")
+
+    {:ok, _} =
```

Comment:
1 line + explain ttrick. In fact scan through all the comments in the other tests, many apply here too

Reply:
Same as in another comment. I clicked resolve here to follow only one conversation.